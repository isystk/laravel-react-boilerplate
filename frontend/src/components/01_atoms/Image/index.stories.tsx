import { Meta, Story } from '@storybook/react'
import React from 'react'
import Image, { ImageProps } from './index'

export default {
  title: '01_atoms/Image',
  component: Image,
} as Meta

const Template: Story = () => {
  const props: ImageProps = {
    src: '/images/dummy_480x320.png',
    alt: 'dummy',
    zoom: true,
  }
  return (
    <div className="w-64 rounded-lg overflow-hidden">
      <Image {...props} />
    </div>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
