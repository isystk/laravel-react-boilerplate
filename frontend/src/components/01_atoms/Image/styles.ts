import { css, keyframes } from '@emotion/css'

export const image = css`
  width: 100%;
  height: 100%;
`
