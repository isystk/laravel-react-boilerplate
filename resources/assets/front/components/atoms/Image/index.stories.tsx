import type { Meta, StoryFn } from '@storybook/react';
import Image from './index';

export default {
  title: 'Components/Atoms/Image',
  component: Image,
  tags: ['autodocs'],
} as Meta<typeof Image>;

export const Default: StoryFn = () => (
  <Image
    src="/assets/images/profile.jpeg"
    alt="サンプル画像"
    width={200}
    height={100}
    className="border"
  />
);

export const WithLazyLoading: StoryFn = () => (
  <Image
    src="/assets/images/profile.jpeg"
    alt="遅延読み込み画像"
    loading="lazy"
    width={300}
    height={150}
  />
);

export const WithEagerLoading: StoryFn = () => (
  <Image
    src="/assets/images/profile.jpeg"
    alt="即時読み込み画像"
    loading="eager"
    width={300}
    height={150}
  />
);
