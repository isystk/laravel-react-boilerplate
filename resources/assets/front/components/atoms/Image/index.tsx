type Props = {
  src: string;
  alt?: string;
  width?: number;
  height?: number;
  loading?: 'eager' | 'lazy';
  className?: string;
};

const Image = ({ src, alt = '', loading = 'lazy', ...props }: Props) => {
  const subDirectory = import.meta.env.SUB_DIRECTORY || '';
  return <img src={`${subDirectory}${src}`} alt={alt} loading={loading} {...props} />;
};

export default Image;
