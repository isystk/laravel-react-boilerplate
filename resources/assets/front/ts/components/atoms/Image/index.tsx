
type Props = {
    src: string,
    alt?: string,
    width: number,
    height: number,
    loading?: "eager" | "lazy",
    className?: string
};

const Image = ({ src, alt, loading, ...props }: Props) => {
    const subDirectory = import.meta.env.SUB_DIRECTORY || "";
    return (
        <>
            <img
                src={`${subDirectory}${src}`}
                alt={alt || ''}
                loading={loading || 'lazy'}
                {...props}
            />
        </>
    );
};

export default Image;

