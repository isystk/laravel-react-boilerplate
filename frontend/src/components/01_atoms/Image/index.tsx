import React, { FC } from 'react'
import { ContainerProps } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** ImageProps Props */
export type ImageProps = {
  src: string
  alt: string
  className?: string
  lazy?: boolean
  zoom?: boolean
}
/** Presenter Props */
export type PresenterProps = ImageProps

/** Presenter Component */
const ImagePresenter: FC<PresenterProps> = ({
  src,
  alt,
  className,
  lazy = true,
  zoom = false,
}) => (
  <div className={className}>
    <img
      src={src}
      alt={alt}
      className={`${styles.image} ${zoom ? 'zoom' : ''}`}
      loading={lazy ? 'lazy' : undefined}
    />
  </div>
)

/** Container Component */
const ImageContainer: React.FC<ContainerProps<ImageProps, PresenterProps>> = ({
  presenter,
  ...props
}) => {
  return presenter({
    ...props,
  })
}

export default connect<ImageProps, PresenterProps>(
  'Image',
  ImagePresenter,
  ImageContainer
)
