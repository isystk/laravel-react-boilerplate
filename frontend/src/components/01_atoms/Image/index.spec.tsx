import React from 'react'
import renderer from 'react-test-renderer'
import Image, { ImageProps } from './index'

describe('Image', () => {
  it('Match Snapshot', () => {
    const props: ImageProps = {
      src: '/images/dummy_480x320.png',
      alt: 'dummy',
      zoom: true,
    }
    const component = renderer.create(
      <div className="w-64 rounded-lg overflow-hidden">
        <Image {...props} />
      </div>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
    expect(true)
  })
})
