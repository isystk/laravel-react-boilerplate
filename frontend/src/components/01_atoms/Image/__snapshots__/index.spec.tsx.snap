// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Match Snapshot 1`] = `
<div
  className="w-64 rounded-lg overflow-hidden"
>
  <div>
    <img
      alt="dummy"
      className="css-s08p0c zoom"
      loading="lazy"
      src="/images/dummy_480x320.png"
    />
  </div>
</div>
`;
